sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedFunctions.scala

```diff
@@ -78,37 +80,52 @@ object ResolveTableValuedFunctions extends Rule[LogicalPlan] {
   private val builtinFunctions: Map[String, TVF] = Map(
     "range" -> Map(
       /* range(end) */
-      tvf("end" -> LongType) { case Seq(end: Long) =>
-        Range(0, end, 1, None)
+      tvf("end" -> LongType) { case (tvf, args @ Seq(end: Long)) =>
+        validateInputDimension(tvf, 1)
+        Range(0, end, 1, None, tvf.outputNames.headOption)
```

Comment:
Instead of changing the output of `Range`, I think we can simply add a `Project` above Range for column alias, like what we do in the Dataframe API: `spark.range(100).toDF("i")`